### PR TITLE
Derive from runtime_error for pyOpenMS error messages

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Exceptions don't seem to be properly propagated to pyOpenMS. Deriving our exception
classes from runtime_error forwards exception messages to python.
See: https://stackoverflow.com/questions/10684983/handling-custom-c-exceptions-in-cython

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
```
  const Residue& AASequence::getResidue(Size index) const
  {
    if (index >= peptide_.size())
    {
      throw Exception::IndexOverflow(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, index, peptide_.size());
    }
    return *peptide_[index];
  }
```
now throws on python side:
```

Traceback (most recent call last):
  File "/home/sachsenb/miniconda3/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/sachsenb/OMS/OpenMS/openms-build/pyOpenMS/tests/unittests/test000.py", line 32, in wrapper
    f(*a, **kw)
  File "/home/sachsenb/OMS/OpenMS/openms-build/pyOpenMS/tests/unittests/test000.py", line 227, in testAASequence
    seq.getResidue(100) # test exception
  File "pyopenms/pyopenms_6.pyx", line 264, in pyopenms.pyopenms_6.AASequence.getResidue
RuntimeError: the given index was too large: 100 (size = 16)
```

File: src/openms/source/CONCEPT/Exception.cpp [hunk 1]
```diff
@@ -56,42 +57,41 @@ namespace OpenMS
...
-      name_(name),
-      what_(message)
+      name_(name)
     {
-      GlobalExceptionHandler::getInstance().set(file_, line_, function_, name_, what_);
+      GlobalExceptionHandler::getInstance().set(file_, line_, function_, name_, this->what());
     }
 
     BaseException::BaseException(const char* file, int line, const char* function) noexcept :
+      std::runtime_error("unknown error"),
```
Comment (jpfeuffer) [inline src/openms/source/CONCEPT/Exception.cpp:80]: Did you specifically want to make a distinction here to the default ctor?

**Review by jpfeuffer — APPROVED** [review]

File: src/openms/source/CONCEPT/Exception.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/source/CONCEPT/Exception.cpp:80]: no I just used the messages that were there.
I think I can make both unknown error?

Comment (jpfeuffer) [reply src/openms/source/CONCEPT/Exception.cpp:80]: ah hmm.. don't care much. Maybe it gives a bit more hints where the error came from, this way.

**Pull request merged**